lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -23,46 +22,54 @@ class HtmlCodec implements ui.Codec {
 
   @override
   Future<ui.FrameInfo> getNextFrame() async {
-    StreamSubscription<html.Event> loadSubscription;
-    StreamSubscription<html.Event> errorSubscription;
     final Completer<ui.FrameInfo> completer = Completer<ui.FrameInfo>();
-    final html.ImageElement imgElement = html.ImageElement();
-    // If the browser doesn't support asynchronous decoding of an image,
-    // then use the `onload` event to decide when it's ready to paint to the
-    // DOM. Unfortunately, this will case the image to be decoded synchronously
-    // on the main thread, and may cause dropped framed.
-    if (!_supportsDecode) {
-      loadSubscription = imgElement.onLoad.listen((html.Event event) {
-        loadSubscription.cancel();
-        errorSubscription.cancel();
+    if (_supportsDecode) {
+      final html.ImageElement imgElement = html.ImageElement();
+      imgElement.src = src;
+      js_util.setProperty(imgElement, 'decoding', 'async');
+      imgElement.decode().then((dynamic _) {
         final HtmlImage image = HtmlImage(
           imgElement,
           imgElement.naturalWidth,
           imgElement.naturalHeight,
         );
         completer.complete(SingleFrameInfo(image));
+      }).catchError((e) {
+        // This code path is hit on Chrome 80.0.3987.16 when too many
+        // images are on the page (~1000).
+        // Fallback here is to load using onLoad instead.
+        _decodeUsingOnLoad(completer);
```

Comment:
I think a better strategy might be to have a configurable limit on decoding concurrency. We'd then just pick a limit per browser. That may not obviate the need for this fallback, but it would be good to not rely on the fallback during normal app execution.

Reply:
Even if we set a limit for Chrome and monitored rate of decoding it could change over time with no way to detect. Since it is edge case and likely to be fixed (other js image libraries reverted usage, its a known issue), fallback seemed best option.

Reply:
SGTM